gh-badges/lib/index.js

```diff
@@ -28,7 +29,17 @@ function _validate(format) {
     }
   })
 
-  const formatValues = ['svg', 'json']
+  try {
+    require('gm')
+  } catch (e) {
+    if ('format' in format && ['png', 'jpg', 'gif'].includes(format.format)) {
+      throw new ValidationError(
+        `peerDependency gm is required for output in .${format.format} format`
```

Comment:
I'm not really sure how to test this case.
I've moved `gm` (which is now a `peerDependency`) to a dev dependency of the application. That means we can run the tests for raster formats, but then it means we can't test this case. The 2 methods I can see would be:

- Somehow mock the call to require to force it to throw
- Hive off this one test and run it in its own CI build without `gm` installed (and remove it from the main `npm test` run)

I don't really have any experience testing stuff with peer dependencies. Anyone know what is the usual tactic for this? In one sense, the simple approach is just to ignore this and go with the approach in #4523 , but its concept we'll need to find an approach for if we expose logos in the library and make `simple-icons` a peer dependency anyway.

Reply:
I wonder if using [require.resolve](https://nodejs.org/api/modules.html#modules_require_resolve_request_options) would be easier to test than a try/catch 🤔 

```js
  if (!require.resolve('gm')) {
    // conditional format check and throw
  }
```

and then in the unit tests use Sinon to stub conditionally?
```js
  Sinon.stub(require, 'resolve').withArgs('gm').callsFake(() => ...)
```

Disclaimer though, haven't tried that and have absolutely no clue if that'll actually work

Reply:
did a bit of reading and the consensus seems to be: use proxyquire for this
https://github.com/thlorenz/proxyquire#using-proxyquire-to-simulate-the-absence-of-modules